### PR TITLE
Uplifted eiffel-remrem-semantics version from 2.0.6 to 2.0.8.

### PR DESCRIPTION
**Pull request by zkxxdiv**
Handled NullPointerException for Null cases.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 2.0.11
```
Comment (SantoshNC68) [inline CHANGELOG.md]: this should not be uplifted, we haven't released 2.0.10 yet. 

Add this to 2.0.10

File: pom.xml [hunk 2]
```diff
@@ -10,7 +10,7 @@
     </parent>
 
     <properties>
-        <eiffel-remrem-generate.version>2.0.10</eiffel-remrem-generate.version>
+        <eiffel-remrem-generate.version>2.0.11</eiffel-remrem-generate.version>
```
Comment (SantoshNC68) [inline pom.xml]: keep it to 2.0.10

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/constants/RemremGenerateServiceConstants.java [hunk 3]
```diff
@@ -24,6 +24,8 @@
 
         public static final String INTERNAL_SERVER_ERROR = "{\"status_code\": 500, \"result\": \"FAIL\", "
                 + "\"message\":\"Internal server error\"}";
+        public static final String  JSON_ERROR= "{\"status_code\": 400, \"result\": \"FAIL\", "
```
Comment (SantoshNC68) [inline service/src/main/java/com/ericsson/eiffel/remrem/generate/constants/RemremGenerateServiceConstants.java]: MISSING_FIELDS_ERROR

File: service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java [hunk 4]
```diff
@@ -121,6 +121,10 @@ public void setRestTemplate(RestTemplate restTemplate) {
             } else {
                 return new ResponseEntity<>(parser.parse(e1.getMessage()), HttpStatus.EXPECTATION_FAILED);
             }
+        } catch (NullPointerException e) {
+            log.error("Input json should be proper", e);
```
Comment (e-pettersson-ericsson) [inline service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java]: json --> JSON

**Comment by magnusbaeck** [comment]
I think this correction is made in the wrong place. The correct place would be in generateMsg() where the NPE originates. Reasons:

* We can't blindly assume that any NPE resulting from the call to generateMsg() is caused by a value missing from the input.
* Only in generateMsg() can we produce an error message that includes information about what value is missing.
* eiffel-semantics has other clients that would benefit from a correction there.
* A bad message type value is caught by generateMsg() so it's reasonable that it checks other input values too.

**Comment by zkxxdiv** [comment]
> I think this correction is made in the wrong place. The correct place would be in generateMsg() where the NPE originates. Reasons:
> 
> * We can't blindly assume that any NPE resulting from the call to generateMsg() is caused by a value missing from the input.
> * Only in generateMsg() can we produce an error message that includes information about what value is missing.
> * eiffel-semantics has other clients that would benefit from a correction there.
> * A bad message type value is caught by generateMsg() so it's reasonable that it checks other input values too.


Hi Magnus Bäck,
Handled the NPE where it is originates that is REMReM semantics. Please have a look on PR : https://github.com/eiffel-community/eiffel-remrem-semantics/pull/121.

**Comment by zkxxdiv** [comment]
By mistake pull request got closed

**Pull request merged**